fastfold/model/fastnn/evoformer.py

```diff
@@ -108,15 +108,18 @@ def inplace(
         if self.first_block:
             m[0] = m[0].unsqueeze(0)
             z[0] = z[0].unsqueeze(0)
+            torch.cuda.empty_cache()
```

Comment:
empty_cache slows down the speed significantly, maybe its better to add a flag?

Reply:
use drop_unused and torch.cuda.empty_cache() will slows down the speed, but I test it can inference with lower memory usage.
[test_scatter.zip](https://github.com/hpcaitech/FastFold/files/10741151/test_scatter.zip)

Reply:
I have tried your code. Here is the question:
The`torch.cuda.max_memory_allocated()` increased, which means it makes the peak memory increasing.
I think your motivation idea is: with `.clone()` and `torch.cuda.empty_cache()` to escape `tensor_list`, only keep `tensor_list[rank]` to save memory. However, when `.clone()`, there are both `tensor_list` and `output` in the memory, leading to an increase of peak memory.